haiku-apps/beae/beae-1.2.recipe

```diff
@@ -5,9 +5,10 @@ HOMEPAGE="https://github.com/HaikuArchives/BeAE"
 COPYRIGHT="2003 Xentronix"
 LICENSE="BSD (3-clause)"
 REVISION="4"
-SOURCE_URI="https://github.com/HaikuArchives/BeAE/archive/v1.2.tar.gz"
+SOURCE_URI="https://github.com/HaikuArchives/BeAE/archive/v$portVersion.tar.gz"
 CHECKSUM_SHA256="d69a16821f86eda6e94b8297928f9def75f5c750e1e7d6b946d0b195fdc88943"
-SOURCE_DIR="BeAE-1.2"
+SOURCE_DIR="BeAE-$portVersion"
+SOURCE_FILENAME="BeAE-$portVersion.tar.gz"
 
 ARCHITECTURES="x86_gcc2 !x86 ?x86_64"
```

Comment:
@scottmc, I just tried to build the recipe on x86_64 but it fails with this error:
```
MyClipBoard.cpp: In member function 'void MyClipBoard::Update()':
MyClipBoard.cpp:697:41: error: cast from 'void*' to 'team_id {aka int}' loses precision [-fpermissive]
   team_id who = (team_id)teams->ItemAt(i);
                                         ^
make: *** [/boot/system/develop/etc/makefile-engine:291: objects/MyClipBoard.o] Error 1
```
So I guess we should probably change the "?x86_64" to "!x86_64".

While at it, could you please swap the lines for SOURCE_DIR and SOURCE_FILENAME?
I had **incorrectly** suggested inserting SOURCE_FILENAME after SOURCE_DIR, while the [HaikuPorter guidelines](https://github.com/haikuports/haikuports/wiki/HaikuPorter-Guidelines#ordering) say that SOURCE_FILENAME comes first. I can't explain why I swapped it... Sorry, and thanks!

Oops, the guidelines do not say anything about the SOURCE_FILENAME variable!
It's just in a small template I had made for my own use, because I had noticed that:
```
SOURCE_URI
CHECKSUM_SHA256
SOURCE_FILENAME
SOURCE_DIR
PATCHES
ADDITIONAL_FILES
```
was the most frequent order. I guess we should mention SOURCE_FILENAME in the guidelines.


Regarding the SECONDARY_ARCHITECTURES, maybe we could remove the "!x86" and add a "x86_gcc2". I'll try this.